examples/lambda/barendregt/standardisationScript.sml

```diff
@@ -2499,10 +2488,68 @@ Proof
      MATCH_MP_TAC LAMl_SUB \\
      Q.PAT_X_ASSUM ‘closed N’ MP_TAC >> rw [closed_def])
  >> DISCH_TAC
- >> MATCH_MP_TAC lameq_trans
+ >> MATCH_MP_TAC lameq_TRANS
  >> Q.EXISTS_TAC ‘LAMl vs ([N/v] M) @* Ns’ >> art []
  >> MATCH_MP_TAC lameq_appstar_cong >> art []
 QED
 
+Theorem hnf_LAMl[simp] :
+    hnf (LAMl vs M) <=> hnf M
+Proof
+    Induct_on ‘vs’ >> rw []
+QED
+
+Theorem hnf_appstar :
```

Comment:
If you lift the `Ns <> []` assumption out, I believe the statement could be an iff:
```
Ns <> [] ==> (hnf (M @* Ns) <=> hnf M /\ ~is_abs M)
```